From: Developer
Date: Thu, 06 Aug 2026
Subject: [PATCH] add getSSID stub to WifiInfo to prevent F-Droid crashes

F-Droid crashes because it expects `getSSID()` to exist in `android.net.wifi.WifiInfo`.
Return `"<unknown ssid>"` which is the default for AOSP when SSID is not available.
API: https://developer.android.com/reference/android/net/wifi/WifiInfo#getSSID()

---
 src/api-impl/android/net/wifi/WifiInfo.java | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/api-impl/android/net/wifi/WifiInfo.java b/src/api-impl/android/net/wifi/WifiInfo.java
index 1111111..2222222 100644
--- a/src/api-impl/android/net/wifi/WifiInfo.java
+++ b/src/api-impl/android/net/wifi/WifiInfo.java
@@ -9,4 +9,8 @@ public class WifiInfo {
 	public String getBSSID() {
 		return "";
 	}
+
+	public String getSSID() {
+		return "<unknown ssid>";
+	}
 }
